src/sentry/workflow_engine/metric_helpers.py

```diff
@@ -0,0 +1,166 @@
+from sentry.incidents.grouptype import MetricAlertFire
+from sentry.incidents.models.alert_rule import (
+    AlertRule,
+    AlertRuleThresholdType,
+    AlertRuleTrigger,
+    AlertRuleTriggerAction,
+)
+from sentry.snuba.models import QuerySubscription
+from sentry.users.services.user import RpcUser
+from sentry.workflow_engine.models import (
+    Action,
+    AlertRuleDetector,
+    AlertRuleTriggerDataCondition,
+    AlertRuleWorkflow,
+    DataCondition,
+    DataConditionGroup,
+    DataConditionGroupAction,
+    DataSource,
+    Detector,
+    DetectorState,
+    DetectorWorkflow,
+    Workflow,
+    WorkflowDataConditionGroup,
+)
+from sentry.workflow_engine.models.data_condition import Condition, ConditionType
+from sentry.workflow_engine.types import ActionType, DataSourceType, DetectorPriorityLevel
+
+
+def create_metric_action(alert_rule_trigger_action: AlertRuleTriggerAction) -> None:
+    try:
+        alert_rule_trigger_data_condition = AlertRuleTriggerDataCondition.objects.get(
+            alert_rule_trigger=alert_rule_trigger_action.alert_rule_trigger
+        )
+    except AlertRuleTriggerDataCondition.DoesNotExist:
+        return None
+
+    data = {
+        "type": alert_rule_trigger_action.type,
```

Comment:
@saponifi3d we are adding the integration "provider" to the data blob here. my current plan is to not use this and deduce it via `integration_id` and `target_type`.  i am still open to switching over and using this "type" if you think its a good idea for future work.

this might also make the UI portion of ACI a little easier since we will have a `provider`

Reply:
if we split up the types per integration provider, would we still then have a split between "issue" and "metric" as we discussed yesterday?

Reply:
discussed in office hours: we will add a legacy field on an `Action` for "issue" and "metric"